File: Projekt1/frontend/src/types/Item.ts
export interface Item {
  id: number;
  name: string;
  description: string;
  created_at: string;
  owner_id: number;
}

export interface ItemCreate {
  name: string;
  description: string;
}

export interface ItemUpdate {
  name?: string;
  description?: string;
}

File: Projekt1/backend/app/schemas/item.py
from datetime import datetime

from pydantic import BaseModel


class ItemOut(BaseModel):
    id: int
    name: str
    description: str
    created_at: datetime
    owner_id: int

    model_config = {"from_attributes": True}


class ItemCreate(BaseModel):
    name: str
    description: str = ""


class ItemUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
